File: CameraColor/src/main/java/org/caojun/cameracolor/activity/WelcomeActivity.java
package org.caojun.cameracolor.activity;

import android.Manifest;
import android.os.Bundle;
import com.alibaba.android.arouter.launcher.ARouter;
import org.caojun.activity.BaseActivity;
import org.caojun.cameracolor.Constant;
import org.caojun.cameracolor.R;
import org.caojun.utils.ActivityUtils;

/**
 * Created by CaoJun on 2018-3-23.
 */

public class WelcomeActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        this.checkSelfPermission(Manifest.permission.CAMERA, new ActivityUtils.RequestPermissionListener() {

            @Override
            public void onFail() {
                finish();
            }

            @Override
            public void onSuccess() {
                ARouter.getInstance().build(Constant.ACTIVITY_MAIN).navigation();
                finish();
            }
        });
    }
}

File: TTSchulte/src/main/java/org/caojun/ttschulte/activity/WelcomeActivity.kt
package org.caojun.ttschulte.activity

import android.app.Activity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_welcome.*
import org.caojun.particle.ParticleView
import org.caojun.ttschulte.R
import org.jetbrains.anko.startActivity

class WelcomeActivity: Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_welcome)

        particleView.startAnim()
        particleView.setOnParticleAnimListener(object : ParticleView.ParticleAnimListener {
            override fun onAnimationEnd() {
                startActivity<GameActivity>()
                finish()
            }
        })
    }
}